File: src/edu/gbcg/dbInteraction/dbcreator/reddit/comments/CommentsFacilitator.java
package edu.gbcg.dbInteraction.dbcreator.reddit.comments;

import edu.gbcg.configs.DBLocator;
import edu.gbcg.configs.Finals;
import edu.gbcg.configs.RawDataLocator;
import edu.gbcg.configs.columnsAndKeys.RedditComments;
import edu.gbcg.dbInteraction.dbcreator.reddit.Facilitator;
import edu.gbcg.dbInteraction.dbcreator.reddit.JsonPusher;

import java.util.ArrayList;
import java.util.List;

public class CommentsFacilitator extends Facilitator{
    public CommentsFacilitator(){ super(); }

    protected List<String> buildDBPaths(){
        return DBLocator.buildComDBPaths();
    }

    protected List<String> getJsonAbsolutePaths(){
        return RawDataLocator.redditJsonCommentAbsolutePaths();
    }

    protected List<String> getDBAbsolutePaths(){
        return DBLocator.redditComsAbsolutePaths();
    }

    protected List<String> getDBDirectoryPaths(){
        return DBLocator.getComDBPath();
    }

    protected List<String> getJsonKeysOfInterest(){
        return RedditComments.JSONkeys();
    }

    protected List<String> getColumnNames(){
        return RedditComments.columnNames();
    }

    protected List<String> getDataTypes(){
        return RedditComments.dataTypes();
    }

    protected List<JsonPusher> populateJsonWorkers(){
        List<JsonPusher> workers = new ArrayList<>();
        for(int i = 0; i < Finals.DB_SHARD_NUM; ++i)
            workers.add(new CommentsJsonPusher());
        return workers;
    }

    protected String getTableName(){
        return Finals.COM_TABLE_NAME;
    }

    protected void createIndices(){
        createDBIndex(Finals.AUTHOR, "attrs_author");
        createDBIndex("controversial_score", "attrs_cont_score");
        createDBIndex(Finals.CREATED_DT, "attrs_created");
        createDBIndex("gilded", "attrs_gilded");
        createDBIndex(Finals.POST_ID, "attrs_pid");
        createDBIndex("link_id", "attrs_linkid");
        createDBIndex("parent_id", "attrs_parentid");
        createDBIndex(Finals.SCORE, "attrs_score");
        createDBIndex("subreddit_name", "attrs_sub_name");
        createDBIndex("subreddit_id", "attrs_sub_id");
    }
}

File: src/main/kotlin/edu/gbcg/dbInteraction/dbcreator/reddit/comments/CommentsFacilitator.kt
/*
 * Copyright (c) 2018 Sean Grimes. All Rights Reserved.
 * License: MIT
 */

package edu.gbcg.dbInteraction.dbcreator.reddit.comments

import edu.gbcg.configs.DBLocator
import edu.gbcg.configs.Finals
import edu.gbcg.configs.RawDataLocator
import edu.gbcg.configs.columnsAndKeys.RedditComs
import edu.gbcg.dbInteraction.dbcreator.Facilitator
import edu.gbcg.dbInteraction.dbcreator.reddit.JsonPusher

@Suppress("ConvertSecondaryConstructorToPrimary")
class CommentsFacilitator: Facilitator {
    constructor(): super()

    override fun buildDBPaths()                     = DBLocator.buildComDBPaths()
    override fun getJsonAbsolutePaths()             = RawDataLocator.redditJsonCommentAbsolutePaths()
    override fun getDBAbsolutePaths()               = DBLocator.redditComsAbsolutePaths()
    override fun getDBDirectoryPaths()              = DBLocator.getComDBPath()
    override fun getJsonKeysOfInterest()            = RedditComs.JSONKeys()
    override fun getColumnNames()                   = RedditComs.columnNames()
    override fun getDataTypes()                     = RedditComs.dataTypes()
    override fun getTableName()                     = Finals.COM_TABLE_NAME
    override fun getJsonAbsolutePathsForNewData()   = RawDataLocator.redditJsonCommentAbsolutePathsNewData()

    override fun populateJsonWorkers(): List<JsonPusher> {
        val workers = ArrayList<JsonPusher>()
        for(i in 0 until Finals.DB_SHARD_NUM)
            workers.add(CommentsJsonPusher())
        return workers
    }

    override fun createIndices() {
        createDBIndex(Finals.AUTHOR, "attrs_author")
        createDBIndex("controversial_score", "attrs_cont_score")
        createDBIndex(Finals.CREATED_DT, "attrs_created")
        createDBIndex("gilded", "attrs_gilded")
        createDBIndex(Finals.POST_ID, "attrs_pid")
        createDBIndex("link_id", "attrs_linkid")
        createDBIndex("parent_id", "attrs_parentid")
        createDBIndex(Finals.SCORE, "attrs_score")
        createDBIndex("subreddit_name", "attrs_sub_name")
        createDBIndex("subreddit_id", "attrs_sub_id")
    }

    override fun dropIndices() {
        dropDBIndices("attrs_author")
        dropDBIndices("attrs_cont_score")
        dropDBIndices("attrs_created")
        dropDBIndices("attrs_gilded")
        dropDBIndices("attrs_pid")
        dropDBIndices("attrs_linkid")
        dropDBIndices("attrs_parentid")
        dropDBIndices("attrs_score")
        dropDBIndices("attrs_sub_name")
        dropDBIndices("attrs_sub_id")
    }
}
